### PR TITLE
shared control plane multicluster fixes

### PR DESCRIPTION
**Pull request by ayj**
* Rename the remote istiod service and endpoint to `istiod-remote` to
  avoid conflicts with the real local istiod service.

* Use the `istiod-remote.<namespace>.svc` hostname for the sidecar and
  ingress proxies discoveryAddress. This address needs to match the
  SAN in istiod's cert. The `istiod-remote` headless service will
  resolve the hostname to the remote IP address.

* Add the `istiod-remote` hostname to istiod's SANs. Also use istiod's
  namespace to construct the legacy service names instead of
  hardcoding them to `istio-system`.

* Simplify the remote profile by removing redundant and unused values.

Manually backport https://github.com/istio/istio/pull/21912 from master.

### CODE REVIEW

**Comment by ayj** [comment]
cc @linsun

File: istioctl/pkg/multicluster/remote_secret.go [hunk 1]
```diff
@@ -86,15 +86,15 @@ func NewCreateRemoteSecretCommand() *cobra.Command {
...
 
 # Delete a secret that was previously installed in c1
 istioctl --Kubeconfig=c0.yaml x create-remote-secret --name c0 \
-    | kubectl -n istio-system --Kubeconfig=c1.yaml delete -f -
+    | kubectl --Kubeconfig=c1.yaml delete -f -
 
 # Create a secret access a remote cluster with an auth plugin
 istioctl --Kubeconfig=c0.yaml x create-remote-secret --name c0 --auth-type=plugin --auth-plugin-name=gcp \
-    | kubectl -n istio-system --Kubeconfig=c1.yaml apply -f -
+    | kubectl --Kubeconfig=c1.yaml apply -f -
```
Comment (linsun) [inline istioctl/pkg/multicluster/remote_secret.go]: @ayj what would be the cmd if users don't install istio to istio-system ns?

Comment (linsun) [reply istioctl/pkg/multicluster/remote_secret.go]: Also, where can I find docs for this auth-plugin, auth-type?

File: pilot/pkg/features/pilot.go [hunk 2]
```diff
@@ -301,4 +301,7 @@ var (
 
 	JwtPolicy = env.RegisterStringVar("JWT_POLICY", jwt.JWTPolicyThirdPartyJWT,
 		"The JWT validation policy.")
+
+	ClusterName = env.RegisterStringVar("CLUSTER_ID", "Kubernetes",
+		"Defines the cluster and service registry that this Istiod instance is belongs to")
```
Comment (linsun) [inline pilot/pkg/features/pilot.go]: is belongs -> belongs

File: istioctl/pkg/multicluster/remote_secret.go (hunk 1 above)
Comment (ayj) [reply istioctl/pkg/multicluster/remote_secret.go]: Yes, user's can use `-n <namespace>` if they install in a different namespace. 

There aren't any additional docs yet for auth->{plugin,type}.

**Comment by ayj** [comment]
This should be ready for review. This is mostly a backport of https://github.com/istio/istio/pull/21912. Charts between 1.5 and master have diverged so its worth taking a second look at the legacy helm charts in particular.

**Pull request merged**